### PR TITLE
Made VGGish and YAMNet work in TF2 without disabling TF2 behavior.

### PR DESCRIPTION
**Pull request by plakal**
Allowed TF2 behavior and allowed passing in a features tensor into the
VGGish model definition. Both of these changes are needed for making
TF-Hub exports of these models. Lifted constraints on TF versions since
tf_slim has been updated to work with TF 2.

# Description

> :memo: Please include a summary of the change. 
>  
> * Please also include relevant motivation and context.  
> * List any dependencies that are required for this change.  

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] Documentation update
- [x] TensorFlow 2 migration

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * Provide instructions so we can reproduce.  
> * Please also list any relevant details for your test configuration.  

**Test Configuration**:

VGGish: Ran vggish_smoke_test.py and vggish_inference_demo.py with latest TF1 and TF2
YAMNet: Only documentation change.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.

### CODE REVIEW

File: research/audioset/vggish/vggish_slim.py [hunk 1]
```diff
@@ -31,28 +31,31 @@
...
-  a 128-D embedding layer, which is usually the penultimate layer when used as
-  part of a full model with a final classifier layer.
+  The input is either a tensor passed in via the optional 'features' argument or
+  a placeholder created below named 'vggish/input_features'. The input is
+  expected to have dtype float32 and shape [batch_size, num_frames, num_bands]
+  where batch_size is variable and num_frames and num_bands are constants, and
+  [num_frames, num_bands] represents a log-mel-scale spectrogram patch covering
+  num_bands frequency bands and num_frames time frames (where each frame step is
+  usually 10ms). This is produced by computing the stabilized
+  log(mel-spectrogram + params.LOG_OFFSET).  The output is a tensor  named
```
Comment (dpwe) [inline research/audioset/vggish/vggish_slim.py:51]: Double space after "tensor".

File: research/audioset/vggish/vggish_slim.py [hunk 2]
```diff
@@ -31,28 +31,31 @@
...
 
 import tensorflow.compat.v1 as tf
-tf.disable_v2_behavior()
 import tf_slim as slim
 
 import vggish_params as params
 
 
-def define_vggish_slim(training=False):
+def define_vggish_slim(features=None, training=False):
```
Comment (dpwe) [inline research/audioset/vggish/vggish_slim.py:39]: When I read this, I assumed it was a way to pass in actual values.  You could call it features_tensor.  In the context of the original function, it was obviously a tensor, but promoted to visibility outside the function, it's more ambiguous.

File: research/audioset/vggish/vggish_slim.py (hunk 1 above)
Comment (plakal) [reply research/audioset/vggish/vggish_slim.py]: Done

File: research/audioset/vggish/vggish_slim.py (hunk 2 above)
Comment (plakal) [reply research/audioset/vggish/vggish_slim.py]: Done

**Pull request merged**